staging/src/k8s.io/code-generator/cmd/validation-gen/validators/skipunimported.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validators
+
+import "k8s.io/apimachinery/pkg/util/sets"
+
+const (
+	skipUnimportedTagName = "k8s:skipUnimported"
+)
+
+type skipUnimported struct{}
+
+func init() {
+	RegisterTagValidator(skipUnimported{})
+}
+
+func (skipUnimported) Init(Config) {}
+
+func (skipUnimported) TagName() string {
+	return skipUnimportedTagName
+}
+
+func (skipUnimported) ValidScopes() sets.Set[Scope] {
+	return sets.New(ScopeField)
+}
+
+func (skipUnimported) GetValidations(context Context, _ []string, _ string) (Validations, error) {
+	return Validations{SkipUnimported: true}, nil
+}
+
+func (skipUnimported) Docs() TagDoc {
+	doc := TagDoc{
+		Tag:    skipUnimportedTagName,
+		Scopes: []Scope{ScopeField},
+		Description: `Indicates that validations declared on the field's type will be skipped when the type is not imported. 
```

Comment:
"Imported" is not a great word - it is confusing wrt Go's `import`.  I used "non-included" elsewhere but it's not much better.

"scope"?  e.g `+k8s:allowOutOfScopeType` ?  Or "OutOfGenScope" ?

Reply:
Should we also throw an error if this is set on a type which IS included?

Reply:
Something doesn't feel right, but I can't figure out what.  Which pkgs are in-scope or not is not intrinsic to the type being validated, it's the flags on the CLI.  So having the type say "it's OK to be non-included" means you get a different result based on flags -- spooky action at a distance?

Perhaps the right formulation is "+k8s:opaqueType" or something -- do not even TRY reach into this type.  That could apply to defaults and anything that walks the type tree.  Would that prevent subfield?  I think no, so maybe "+k8s:dumbType" or something is better?